app-cpp/app355.cpp

```diff
@@ -214,35 +468,46 @@ void PrintTrains(std::list<std::list<OrbitalData>> trainList)
     }
 }
 
+} // namespace anonymous
+
 int main(int argc, char* argv[])
 {
     // measure total time in milliseconds using chrono 
     auto startTotal = std::chrono::high_resolution_clock::now();
 
+    // 4 threads
+    SatOrbit satOrbit{4};
+
     // meaure time for each section in milliseconds using chrono
     auto start1 = std::chrono::high_resolution_clock::now();
-    std::vector<sat355::TLE> tleList{ ReadFromFile(argc, argv) };
+    std::vector<sat355::TLE> tleVector{ satOrbit.ReadFromFile(argc, argv) };
     auto end1 = std::chrono::high_resolution_clock::now();
     std::chrono::duration<double, std::milli> elapsed1 = end1 - start1;
     std::cout << "Read from file: " << elapsed1.count() << " ms" << std::endl;
 
     auto start2 = std::chrono::high_resolution_clock::now();
-    std::list<OrbitalData> orbitalList{ CalculateOrbitalData(tleList) };
+    std::vector<OrbitalData> orbitalVector{ satOrbit.CalculateOrbitalData(tleVector) };
     auto end2 = std::chrono::high_resolution_clock::now();
     std::chrono::duration<double, std::milli> elapsed2 = end2 - start2;
     std::cout << "Calculate orbital data: " << elapsed2.count() << " ms" << std::endl;
 
     auto start3 = std::chrono::high_resolution_clock::now();
-    std::list<std::list<OrbitalData>> trainList{ CreateTrains(orbitalList) };
+    satOrbit.SortOrbitalVector(orbitalVector);
     auto end3 = std::chrono::high_resolution_clock::now();
     std::chrono::duration<double, std::milli> elapsed3 = end3 - start3;
```

Comment:
in a future PR...
Consider creating a class for "timer"
Where your timer class has a `Start()` and `NowMSec()` methods.
then you can eliminate all the `std::chrono` boilerplate code